### PR TITLE
chore: Make Go Pipeline better and safer

### PR DESCRIPTION
**Pull request by szepeviktor**
Please see comments below.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -20,9 +20,9 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v4
       with:
-        go-version: 1.21
+        go-version: '1.21'
```
Comment (szepeviktor) [inline .github/workflows/go.yml:23]: YAML parses `1.20` as a float = 1.2

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -34,13 +34,13 @@ jobs:
...
         
     - name: Static Analysis
       run: go vet ./...
 
     - name: Check Formatting
-      run: if [ "$(gofmt -s -l -e . | wc -l)" -gt 0 ]; then exit 1; fi
+      run: test -z "$(gofmt -s -l -e .)"
```
Comment (szepeviktor) [inline .github/workflows/go.yml:43]: voilà

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -51,10 +51,12 @@ jobs:
...
       uses: actions/setup-go@v4
       with:
-        go-version: 1.21
+        go-version: '1.21'
 
     - name: Build
-      run: CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags='-w -s -extldflags "-static"' -tags netgo -o validator cmd/validator/validator.go
+      run: |
+        CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
+          go build -ldflags='-w -s -extldflags "-static"' -tags netgo -o validator cmd/validator/validator.go
```
Comment (szepeviktor) [inline .github/workflows/go.yml:59]: - quotes are not YAML friendly
- make it readable

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -69,29 +71,28 @@ jobs:
...
-        go-version: 1.21
+        go-version: '1.21'
 
     - name: Unit test
       run: go test -v -cover -coverprofile coverage.out ./...
 
     - name: Check coverage
       id: check-coverage
+      env:
+        COVERAGE_THRESHOLD: 95
```
Comment (szepeviktor) [inline .github/workflows/go.yml:82]: chronological order

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -100,31 +101,31 @@ jobs:
...
         # Update README.md and index.md
         for markdown_file in README.md index.md; do
-          sed -i "/id=\"cov\"/c\\${BADGE_IMG_TAG}" $markdown_file
+          sed -i "/id=\"cov\"/c\\${BADGE_IMG_TAG}" "${markdown_file}"
         done
 
         # Check to see if files were updated
-        if [[ `git status --porcelain` ]]; then
-          echo "badge_updates=true" >> "$GITHUB_OUTPUT"
+        if git diff --quiet; then
```
Comment (szepeviktor) [inline .github/workflows/go.yml:113]: there is a got command for that

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -100,31 +101,31 @@ jobs:
...
+          echo "badge_updates=true" >> "${GITHUB_OUTPUT}"
         else
-          echo "badge_updates=false" >> "$GITHUB_OUTPUT"
+          echo "badge_updates=false" >> "${GITHUB_OUTPUT}"
         fi
 
     - name: Commit changes
-      if: steps.generate-badge.outputs.badge_updates == 'true' && github.event_name != 'pull_request'
+      if: steps.generate-badge.outputs.badge_updates == 'true' && github.event_name == 'push'
```
Comment (szepeviktor) [inline .github/workflows/go.yml:120]: be explicit!

File: .github/workflows/go.yml [hunk 7]
```diff
@@ -100,31 +101,31 @@ jobs:
...
       run: |
         git config --local user.email "action@github.com"
         git config --local user.name "GitHub Action"
-        git add README.md index.md
+        git add -- README.md index.md
```
Comment (szepeviktor) [inline .github/workflows/go.yml:124]: distinguish options from pathspec-s

**Review by jd4235 — APPROVED** [review]
LGTM

**Review by kehoecj — APPROVED** [review]
These LGTM - thanks for the PR @szepeviktor !

**Comment by szepeviktor** [comment]
Glad to contribute.

**Comment by szepeviktor** [comment]
@kehoecj Please give me more work. It helps to chase away depression.

**Comment by kehoecj** [comment]
> @kehoecj Please give me more work. It helps to chase away depression.

We'd love to have you contribute to any of our open issues. You can choose from any but here's a few that could be good to get started with:

* https://github.com/Boeing/config-file-validator/issues/61
* https://github.com/Boeing/config-file-validator/issues/64
* https://github.com/Boeing/config-file-validator/issues/38

**Comment by szepeviktor** [comment]
Unfortunately, my expertise is limited to the underlying technologies: everything that makes config-file-validator run.

**Pull request merged**